src/subcommand/primers_main.cpp

```diff
@@ -108,11 +129,36 @@ int main_primers(int argc, char** argv) {
         exit(1);
     }
 
-    string file_name = get_input_file_name(optind, argc, argv);
+    string primers_path = get_input_file_name(optind, argc, argv);
 
-    cout << "primer file name: "      << file_name        << endl
-         << "xg file name: "          << xg_path          << endl
-         << "snarl index file name: " << snarl_index_path << endl;
+    // cout << "primer file name: "      << primers_path     << endl
+    //      << "xg file name: "          << xg_path          << endl
+    //      << "snarl index file name: " << snarl_index_path << endl;
+    
+    SnarlDistanceIndex distance_index;
+    unique_ptr<handlegraph::PathPositionHandleGraph> graph;
+    distance_index.deserialize(snarl_index_path);
+    graph = vg::io::VPKG::load_one<PathPositionHandleGraph>(xg_path);
+    PrimerFinder primer_finder(graph, &distance_index);
+    primer_finder.load_primers(primers_path);
+
+    vector<string> reference_paths = primer_finder.get_reference_paths();
+    for (size_t i = 0; i < reference_paths.size(); ++i) {
+        string path_name = reference_paths[i];
+        const vector<PrimerPair>& primer_pairs = primer_finder.get_primer_pairs_of_chrom(path_name);
+        for (size_t j = 0; j < primer_pairs.size(); ++j) {
+            const PrimerPair& primer_pair = primer_pairs[j];
+            if (all_primers) {
+                print_tabular(path_name, primer_pair);
+            } else if (zero_variance) {
+                if (primer_pair.no_variation_in_products) {
+                    print_tabular(path_name, primer_pair);
+                }
+            } else if (primer_pair.no_variation_at_primers) {
+                print_tabular(path_name, primer_pair);
+            }
```

Comment:
This logic looks wrong to me. Both of the conditions execute only when `primer_pair.no_variation_at_primers`, so the `zero_variance` variable doesn't have any effect.

Reply:
When `zero_variance` is triggered, only primer pairs with no variation at **primers** and in **products** will be printed. Otherwise, primer pairs with no variation at primers would be printed, regardless of variation in products.

Reply:
Ah, right. My mistake.